tests/unit/test_config_node.py

```diff
@@ -345,3 +347,55 @@ def test_deprecation_warnings(
         assert must_include in records
     else:
         assert must_exclude not in records
+
+
+def test_retrieve_installed_cli_opts(tmp_path):
```

Comment:
@oliver-sanders - I've not dotted all the i's or crossed all the t's on this PR - but I'm concerned that this is a fairly fundamental failure - can you check that in your opinion this is failing correctly?

Reply:
Hard to say how this unit should behave without a full review.

However, at a functional level, this:

```
$ cylc vr -S KEY=value
```

Should result in this:

```
$ cylc validate -S KEY=value
$ cylc reinstall -S KEY=value
$ cylc {reload,play}
```

I.E. The prior value of `KEY` should make no difference to command outcome.

Reply:
And presumably if I've changed the source dir that should override the old dir?

Reply:
I think it's the same question.

If you `cylc vr -S KEY=value`, then I would expect `cylc reinstall -S KEY=value`. Because CLI options override config options (they have to), the CLI option wins.

Reply:
OK. Just seems weird. But essentially if you set it by CLI you have to modify it in the same way. That's good, because that's what is happening. 😄

Reply:
> What should happen if you do:
>
> $ cylc install -S KEY=1
> $ printf '[template variables]\nKEY=2\n' > rose-suite.conf
> $ cylc config

In this case you would get `KEY=2`, however, if you did `cylc config --against-source` you would get `KEY=1`.

Reply:
There are four things:

1. New CLI opts
2. Old CLI opts
3. New config
4. Old config

Normally, when we perform an action against a source workflow, we don't have access to the old CLI options / config, we are just looking at the source dir. As a result, we only do 1 & 3.

When we use `--against-source` we are reading the source workflow with the context of the installed workflow. This turns on 2 & 3.